lib/webdriveragent.js

```diff
@@ -149,6 +150,32 @@ class WebDriverAgent {
 
     await B.all(stops);
   }
+
+  //Body must be object
+  async sendCommandWithSession(command, body, method) {
```

Comment:
Hi,

Sorry for the delay, got busy with other work stuff :)

Basically right now the Appium command is called ```background```. It's implemented with a different function name depending on the driver platform I guess, but "background" seems to be the command that ultimately gets passed to the platform driver.

In the case of WebDriverAgent, the event name they created is ```deactivateApp```. So, we need to map the commands. The way I thought of doing it was as you see with this command, where instead of proxying the command I don't proxy it, instead handling it myself in our appium-xuitest-driver code. In our code then I modify it and send it along to WebDriverAgent via the proxy.

I hadn't though that we'd want to modify the generic proxy, since this change would be specific to WebDriverAgent. I suppose I could also try to get Facebook to change their command name; I don't know if they'd want to do this.

Or, another option I guess would be, instead of having to implement custom functions like this, for each case, perhaps the proxy could accept a "mapping" lookup table. Then, for every call the proxy gets, it would check in the mapping lookup table that the appropriate driver provides. If a mapping key is found (e.g. ```{"background":  "deactivateApp"}```) then it changes the command before proxying. It could also handle the body keys too.

What do you think?